hamilton/driver.py

```diff
@@ -1807,6 +1810,34 @@ def with_materializers(
         self.materializers.extend(materializers)
         return self
 
+    def with_cache(
+        self,
+        path: Optional[Union[str, pathlib.Path]] = None,
+        metadata_store: Optional[cache_store.BaseStore] = None,
+        result_store: Optional[cache_store.BaseStore] = None,
+        resume_from: Optional[Union[str, Literal["latest"]]] = None,
+    ) -> "Builder":
+        """Add the caching adapter to the `Driver`
+
+        :param path: path where the cache metadata and results will be stored
+        :param metadata_store: BaseStore handling metadata for the cache adapter
+        :param result_store: BaseStore caching dataflow execution results
+        :param resume_from: Run id or "latest" to load metadata from at execution time.
+        :return: self
+        """
+        metadata_store = (
+            metadata_store if metadata_store else cache_store.SQLiteMetadataStore(path=path)
+        )
+        result_store = result_store if result_store else cache_store.ShelveResultStore(path=path)
+
+        adapter = cache_adapters.SmartCacheAdapter(
+            metadata_store=metadata_store,
+            result_store=result_store,
+            resume_from=resume_from,
```

Comment:
to go with the above:
```suggestion
            **kwargs
```